homeassistant/components/modbus/sensor.py

```diff
@@ -299,10 +359,12 @@ def _update(self):
                 )
         except ConnectionException:
             self._available = False
+            self.schedule_update_ha_state()
```

Comment:
w/o this, the state would be STATE_UNKNOWN instead of STATE_UNAVAILABLE. Added tests as well.

Reply:
Please be aware your PR is about bit_sensor, so please only add changes to that respect.